### PR TITLE
Add client implementation for spark aws

### PR DESCRIPTION
**Pull request by Sami1309**
# Description

Added client methods to register an offline provider via a spark cluster/s3 bucket in AWS

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have fixed any merge conflicts

### CODE REVIEW

**Comment by sdreyer** [comment]
Can this handle dataframe transformations?

**Review by ahmadnazeri — DISMISSED** [review]

File: client/src/featureform/register.py [hunk 1]
```diff
@@ -864,6 +858,31 @@ def get_bigquery(self, name):
         self.__resources.append(get)
         fakeConfig = BigQueryConfig(project_id="", dataset_id="", credentials_path="")
         fakeProvider = Provider(name=name, function="OFFLINE", description="", team="", config=fakeConfig)
+        return OfflineSQLProvider(self, fakeProvider) 
+
+    def get_spark_aws(self, name):
```
Comment (sdreyer) [inline client/src/featureform/register.py:863]: Should this be a generic get_spark?

File: client/src/featureform/register.py [hunk 2]
```diff
@@ -1201,6 +1220,62 @@ def register_bigquery(self,
         self.__resources.append(provider)
         return OfflineSQLProvider(self, provider)
 
+    def register_spark_aws(self,
```
Comment (sdreyer) [inline client/src/featureform/register.py:1223]: Should this be generic register_spark?

File: client/src/featureform/register.py [hunk 3]
```diff
@@ -1201,6 +1220,62 @@ def register_bigquery(self,
...
+                          bucket_path: str = "",
+                          emr_cluster_region: str = "",
+                          bucket_region: str = "",
+                          aws_access_key_id: str = "",
+                          aws_secret_access_key: str = "",):
+        """Register a Spark on AWS provider.
+
+        **Examples**:
+        ```   
+        bigquery = ff.register_spark_aws(
```
Comment (sdreyer) [inline client/src/featureform/register.py:1237]: this has "bigquery"

**Pull request closed without merging**